### PR TITLE
Fix an error when importing text exercises with categories

### LINKED ISSUE #3261: Importing Text Excercises does not work on the production server
### Describe the bug

On production, i am currently not able to import text excercises

#### To Reproduce

1. Go to Course Management
2. Click on your excercises in a Course (used Test: Eist2021 in the screencast)
3. Scroll down to import text excercise
4. Configure all necessary parts (due dates)
5. Scroll down and try to import the excercise
6. See the error message:  Bad request

#### Expected behavior

The excercise should be imported succesfully

#### Screenshots

https://user-images.githubusercontent.com/65229601/115520586-9aba0180-a28a-11eb-8ac4-2601109ff548.mp4

### Environment

<details><pre>
 - OS:Windows 10
 - BrowserChrome
 - Version 90.0.4430.85
</pre></details>

#### Output in Web Console

<details><pre>
</pre></details>

### PR DESCRIPTION
**Pull request by FrankeLukas**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).

### Motivation and Context

Closes #3261.

### Description

The categories weren't serialized properly client side. We now re-use the same methods `create` and `update` use for importing text exercises, too. The commit messages describe the individual changes.

### Steps for Testing

- Create a text exercise with categories
- Update a text exercise with categories
- Import a text exercise with categories

### CODE REVIEW

**Review by cremertim — APPROVED** [review]
Tested on test server 1. Works properly now

File: src/main/webapp/app/exercises/text/manage/text-exercise/text-exercise-update.component.html [hunk 1]
```diff
@@ -205,7 +205,7 @@ <h4 *ngIf="isImport" id="jhi-text-exercise-heading-import" jhiTranslate="artemis
             <fa-icon [icon]="'ban'"></fa-icon>&nbsp;<span jhiTranslate="entity.action.cancel">Cancel</span>
         </button>
 
-        <button type="submit" (click)="save()" class="btn btn-primary" id="submit-entity">
+        <button type="submit" (click)="save()" class="btn btn-primary" id="submit-entity" [disabled]="editForm.form.invalid || isSaving">
```
Comment (Strohgelaender) [inline src/main/webapp/app/exercises/text/manage/text-exercise/text-exercise-update.component.html:208]:
```suggestion
        <button type="submit" (click)="save()" class="btn btn-primary" id="submit-entity" [disabled]="editForm.form.invalid || isSaving || textExercise.dueDateError || textExercise.assessmentDueDateError">
```
I'm not sure if this is needed but this structure is used for the other save-buttons so I think it should be consistent.

**Review by NicolasUllmann — DISMISSED** [review]
Tested on server 1. Everything worked as expected.

File: src/main/webapp/app/exercises/text/manage/text-exercise/text-exercise-update.component.html (hunk 1 above)
Comment (FrankeLukas) [reply src/main/webapp/app/exercises/text/manage/text-exercise/text-exercise-update.component.html]: Good catch, added the extra checks.

**Review by maximilianbergmann — APPROVED** [review]
Tested on TS3. Everything worked especially with categories.

**Review by imstreetspirit — APPROVED** [review]
tested on ts3 as well. works as expected.

**Review by willergeorg — APPROVED** [review]
Tested on TS1 and seems to be working for me.

**Comment by lschlesinger** [comment]
Tested on TS1, works as described

**Review by krusche — APPROVED** [review]

**Pull request merged**